components/addContact/index.test.jsx

```diff
@@ -0,0 +1,233 @@
+import AddContact, {
+  handleSubmit,
+  EXISTING_WEBID_ERROR_MESSAGE,
+  NO_NAME_ERROR_MESSAGE,
+} from "./index";
+import { findContactInAddressBook, saveContact } from "../../src/addressBook";
+import { WithTheme } from "../../__testUtils/mountWithTheme";
+import { defaultTheme } from "../../src/theme";
+import { mockPersonDatasetAlice } from "../../__testUtils/mockPersonResource";
+import { getResource } from "../../src/solidClientHelpers/resource";
+
+jest.mock("@inrupt/solid-client");
+jest.mock("next/router");
+jest.mock("../../src/addressBook");
+jest.mock("../../src/solidClientHelpers/resource");
+
+beforeEach(() => {
+  jest.resetModules();
+});
+
+describe("AddContact", () => {
+  test("it renders the Add Contact form", () => {
+    const session = mockSession();
+    const SessionProvider = mockSessionContextProvider(session);
+    const tree = mount(
+      <SessionProvider>
+        <WithTheme theme={defaultTheme}>
+          <AddContact />
+        </WithTheme>
+      </SessionProvider>
+    );
+
+    expect(mountToJson(tree)).toMatchSnapshot();
+  });
+
+  test("it renders a spinner when user is not logged in", () => {
+    const session = mockUnauthenticatedSession();
+    const SessionProvider = mockSessionContextProvider(session);
+    const tree = mount(
+      <SessionProvider>
+        <WithTheme theme={defaultTheme}>
+          <AddContact />
+        </WithTheme>
+      </SessionProvider>
+    );
+
+    expect(tree).toMatchSnapshot();
+  });
+});
+describe("handleSubmit", () => {
+  test("it alerts the user and exits if the webid already exists", async () => {
+    const personUri = "http://example.com/alice#me";
+    const personDataset = mockPersonDatasetAlice();
+    const setIsLoading = jest.fn();
+    const alertError = jest.fn();
+    const alertSuccess = jest.fn();
+    const handler = handleSubmit({
+      setIsLoading,
+      alertError,
+      alertSuccess,
+      fetch,
+      webId: personUri,
+    });
+    getResource.mockResolvedValue({
+      response: { dataset: personDataset, iri: personUri },
+    });
+    getUrl.mockReturnValue("https://www.example.com/");
+    findContactInAddressBook.mockResolvedValue([personDataset]);
+    await handler();
+    expect(setIsLoading).toHaveBeenCalledTimes(2);
+    expect(alertError).toHaveBeenCalledWith(EXISTING_WEBID_ERROR_MESSAGE);
+  });
+  test("it alerts the user and exits if the webid doesn't have a name", async () => {
+    const { mockThingFrom } = jest.requireActual("@inrupt/solid-client");
+    const personUri = "http://example.com/marie#me";
+    const personDataset = mockThingFrom(personUri);
+    const setIsLoading = jest.fn();
+    const alertError = jest.fn();
+    const alertSuccess = jest.fn();
+    const handler = handleSubmit({
+      setIsLoading,
+      alertError,
+      alertSuccess,
+      fetch,
+      webId: personUri,
+    });
+    getUrl.mockReturnValue("https://www.example.com/");
```

Comment:
You shouldn't need to mock the return value here if you've used the mocking methods from the Solid Client JS.